packages/esm-patient-common-lib/src/workspaces/useWorkspaces.tsx

```diff
@@ -30,10 +33,24 @@ export function useWorkspaces(): WorkspacesInfo {
     }
   }, [workspaces]);
 
-  return {
-    active: workspaces.length > 0,
-    windowState,
-    workspaces,
-    prompt,
-  };
+  useEffect(() => {
+    if (workspaces.length === 0) {
+      updateWorkspaceSizeState('hidden');
+    } else {
+      updateWorkspaceSizeState(workspaces[0].preferredWindowSize === 'maximized' ? 'maximized' : 'normal');
+    }
+  }, [workspaces]);
+
+  const memoisedResults = useMemo(
+    () => ({
+      active: workspaces.length > 0,
+      workspaceWindowState,
+      workspaces,
+      prompt,
+      updateWorkspaceSizeState,
```

Comment:
`updateWorkspaceSizeState` is available via import from `workspaces`. Why would you return it here?

Reply:
This is why I was suggesting making it an action. I’m basically happy as long as we have one clear way of doing this instead of two.

Reply:
I used the `useStoreWithActions`, it helped out well!

Reply:
Actually, my point was to export all the functions (actions) with the state from the `useWorkspaces` only, but was quite doubtful.

@ibacher , your suggestion `useStoreWithActions` really panned out. One issue I found is that the returned action from the hook isn't memoized.